src/polygon.jl

```diff
@@ -0,0 +1,508 @@
+        deviations = complex.(rand(3)*n*4/5 .- n*2/5, rand(3)*m*4/5 .- m*2/5)
+        points = Complex{Int}.(round.(center .+ deviations))
+        # Points are allowed to be outside of canvas as long as part of the triangle is in the canvas
+        # Prevents white borders
+        col = RGB(rand(), rand(), rand())
+        t = Triangle(points[1], points[2], points[3], col)
+        badTriangle = checktriangle(t, m, n)
+    end
+
+    return t
+end
+
+"""
+    generatepopulation(pop_size::Int, number_triangles::Int, img::Array)
+
+Generates an entire population. The individuals of this population consist of:
+- An array of triangles (remember, a triangle is simply 3 xy coordinates of points and a color, not yet an image)
+- The image made from drawing all these triangles onto a white canvas (array of triangles converted to an image)
+- The score of the individual, which is too say how much its image differs from the target image.
+"""
+function generatepopulation(pop_size::Int, number_triangles::Int, img::Array)
+
+    m = length(img[:, 1])
+    n = length(img[1, :])
+    canvas = fill(RGB(1, 1, 1), m, n)
+
+    population = Vector{Array}(undef, pop_size)
+
+    for i in 1:pop_size
+        triangles = [generatetriangle(m, n) for i in 1:number_triangles]
+        polyimg = drawimage(triangles, canvas)
+
+        score = colordiffsum(polyimg, img)
+        population[i] = [triangles, polyimg, score]
+    end
+    return population
+end
+
+
+"""
+    triangletournament(population0::Array, pop_size::Int)
+
+Picks 2 random triangles from a population and returns the one with the best (= lowest) score.
+"""
+function triangletournament(population0::Array, pop_size::Int)
+    contestant1 = population0[Int(ceil(rand()*pop_size))] #Choose a random image from the population, using ceil because 0 is not a valid index
+    contestant2 = population0[Int(ceil(rand()*pop_size))]
+
+    while contestant1 == contestant2 #If it happened to pick the same individual twice, change it until you have 2 unique individuals
+        contestant2 = population0[Int(ceil(rand()*pop_size))]
+    end
+
+    if contestant1[3] <= contestant2[3] # Lower score than contestant 2 means he's better
+        winner = contestant1
+    else
+        winner = contestant2
+    end
+    return winner
+end
+
+"""
+    mutatetriangle(triangle::Triangle, m::Int, n::Int)
+
+Mutates a single triangle, changing the position of its points and its color, while making sure this doesn't result in a stupid triangle.
+"""
+function mutatetriangle(triangle::Triangle, m::Int, n::Int)
+
+    badTriangle = true
+    p1 = nothing
+    p2 = nothing
+    p3 = nothing
+
+    while badTriangle
+        p1, p2, p3 = round.([triangle.p1, triangle.p2, triangle.p3] + (rand(3) .- 0.5)*(n/10) + (rand(3) .- 0.5)*(m/10)*im)
+        # Position of all 3 points is changed, with the change in x and y value scaling with the width and height of the canvas respectively
+        t = Triangle(p1, p2, p3, triangle.color)
```

Comment:
I have a hard time following one-letter variable names. Can `t` be more descriptive?